jax/_src/numpy/lax_numpy.py

```diff
@@ -3706,11 +3706,14 @@ def _index_to_gather(x_shape, idx, normalize_indices=True):
       # Normalize the slice to use None when possible
       start, stop, step = i.start, i.stop, i.step
       try:
-        if ((step is None or core.symbolic_equal_dim(step, 1)) and
-            stop is not None and core.symbolic_equal_dim(stop, x_shape[x_axis])):
-          # The following is a useful special case with shape polymorphism
-          stop = None
-      except TypeError:
+        if step is None or core.symbolic_equal_dim(step, 1):
+          step = None
+        if step is None:
+          if start is None or core.symbolic_equal_dim(start, 0):
+            start = None
+          if stop is None or core.greater_equal_dim(stop, x_shape[x_axis]):
+            stop = None
+      except (TypeError, core.InconclusiveDimensionOperation):
```

Comment:
Handling `InconclusiveDimensionOperation` errors was added since according to the docs `greater_equal_dim` might throw that which isn't the case for `symbolic_equal_dim`.